File: php/features.php
<?php
declare(strict_types=1);

// DymoApiIntroduction SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class DymoApiIntroductionFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new DymoApiIntroductionBaseFeature();
            case "test":
                return new DymoApiIntroductionTestFeature();
            default:
                return new DymoApiIntroductionBaseFeature();
        }
    }
}

File: py/dymoapiintroduction_sdk/features.py
# DymoApiIntroduction SDK feature factory

from dymoapiintroduction_sdk.feature.base_feature import DymoApiIntroductionBaseFeature
from dymoapiintroduction_sdk.feature.test_feature import DymoApiIntroductionTestFeature


def _make_feature(name):
    features = {
        "base": lambda: DymoApiIntroductionBaseFeature(),
        "test": lambda: DymoApiIntroductionTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
